filebeat/crawler/prospector.go

```diff
@@ -35,9 +36,42 @@ func (p *Prospector) Init() error {
 		return err
 	}
 
+	var prospectorer Prospectorer
+
+	switch p.ProspectorConfig.Harvester.InputType {
+	case cfg.StdinInputType:
+		prospectorer = ProspectorStdin{
+			Prospector: p,
```

Comment:
why we need to pass the prospector to prospectorer? Fields 'iteration', 'lastscan', 'registrar' and 'running' should be local to ProspectorStdin or ProspectorLog. All but running are not even required for stdin, right?